diff --git sympy/mpmath/libmp/exec_py3.py sympy/mpmath/libmp/exec_py3.py
index 6b65ae0..41fbd37 100644
--- sympy/mpmath/libmp/exec_py3.py
+++ sympy/mpmath/libmp/exec_py3.py
@@ -1 +1,18 @@
-exec_ = exec
+import sys
+
+from sympy.core.compatibility import PY3, builtins
+
+if PY3:
+    exec_ = getattr(builtins, "exec")
+else:
+    def exec_(_code_, _globs_=None, _locs_=None):
+        """Execute code in a namespace."""
+        if _globs_ is None:
+            frame = sys._getframe(1)
+            _globs_ = frame.f_globals
+            if _locs_ is None:
+                _locs_ = frame.f_locals
+            del frame
+        elif _locs_ is None:
+            _locs_ = _globs_
+        exec("exec _code_ in _globs_, _locs_")
